integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/DataSourceV2RelationVisitor.java

```diff
@@ -31,23 +32,43 @@
 
   private enum Provider {
     ICEBERG,
+    AZURECOSMOS,
```

Comment:
Meanwhile I am working on a PR https://github.com/OpenLineage/OpenLineage/pull/442/files#diff-7b66a9bd5905f4ba42914b73a87d834c1321ebcf75137c1e2a2413c0d85d9db6 that removes all this logic from `DataSourceV2ScanRelationVisitor`. 

I think it is worth being moved to Utils classes and specific handlers per provider as introduced some time ago. In that case, visitors are not affected by each provider's implementation. We are adding plenty of visitors and the less they do the better. 

We need to decide on the ordering of merging branches. I can help with applying this change onto the utils & handlers approach.

Reply:
Thank you @pawel-big-lebowski  for making me aware of your PR!  I'm all for NOT changing the DataSourceV2RelationVisitor. It didn't feel right but it was the only way of getting Cosmos to work in there.

Do you have a plan on when you're going to merge in PR #442 ?

In addition, will you be able to support / overwrite the requirement to determine provider based on the `provider` key in the V2 Data Source's properties?  The Cosmos DB Data Source V2 doesn't use that properties field and so I need to get creative to extract the provider.

Reply:
PR #442 should wait for #447 as this adds `OpenLineageContext` to all visitors and affects multiple files. That should not take long I believe. 

We would need to modify following `CatalogUtils3` methods:
```
  public static Optional<CatalogHandler> getCatalogHandler(TableCatalog catalog) {
  public static Optional<CatalogHandler> getCatalogHandlerByProvider(String provider) {
```
to include table name as argument and check if it starts with certain prefix. Is this right or perhaps azurecosmos uses its own SparkCatalog implementation similar to `org.apache.iceberg.spark.SparkCatalog`? In that case, we would not need to alter `getCatalogHandler` method.

Reply:
@pawel-big-lebowski  - I apologize, I did not see this comment. Yes, the Cosmos connector does appear to have its own [SparkCatalog implementation](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosCatalog.scala) it extends CatalogPlugin     with SupportsNamespaces with TableCatalog.  I'm not as familiar with the Catalog side but would be happy to help integrate it for the Azure Cosmos side when necessary!

Reply:
@wjohnson The PRs #442 and #447 are ready and merged. At least they do not block this feature but still the underlying classes changed and AzureCosmos specific code shall be moved to `AzureCosmosHandler` class (possibly with modifying some handlers methods if necessary). Please let me know if you need help with that?